### PR TITLE
Token's Serialization

### PR DESCRIPTION
**Pull request by Hazurl**

### CODE REVIEW

**Comment by Hazurl** [comment]
Do we wants to move that to another file ?
Do we wants to add the types values ?

File: module.h [hunk 1]
```diff
@@ -479,6 +480,55 @@ namespace ws::module {
 
 
 
+    struct Token {
+        std::string content;
+        std::uint32_t type;
```
Comment (Jackojc) [inline module.h]: Can probably use unsigned char or uint8_t for type, I can't see us using more than 256 different types of tokens.

```suggestion
        std::uint8_t type;
```

File: module.h [hunk 2]
```diff
@@ -479,6 +480,55 @@ namespace ws::module {
...
+        std::uint64_t line;
+        std::uint64_t column;
+    };
+
+    /*
+        8 bytes: content's size
+        <size> bytes: content
+        4 bytes: type
+        8 bytes: line
+        8 bytes: column
```
Comment (Jackojc) [inline module.h]: Maybe we should output the token in the order: 
```
line, column, type, size_content, content
```
to make it in a kind of descending order of size, makes more sense to me.

File: module.h [hunk 3]
```diff
@@ -479,6 +480,55 @@ namespace ws::module {
...
+    */
+    inline std::ostream& serialize(std::ostream& os, Token const& token) {
+        std::size_t size = token.content.size();
+        os.write(reinterpret_cast<const char*>(&size               ), sizeof(size        ));
+        os.write(                               token.content.data(),        size         );
+        os.write(reinterpret_cast<const char*>(&token.type         ), sizeof(token.type  ));
+        os.write(reinterpret_cast<const char*>(&token.line         ), sizeof(token.line  ));
+        os.write(reinterpret_cast<const char*>(&token.column       ), sizeof(token.column));
+        return os;
```
Comment (Jackojc) [inline module.h]: Would it be possible to just reinterpret cast the _entire_ struct into chars and output that?
```suggestion
        os.write(reinterpret_cast<const char*>(&token), sizeof(token))
```

Comment (Hazurl) [reply module.h]: no cause of the content

Comment (Jackojc) [reply module.h]: If you reorder the members so that the primitives come first, you could probably cast them all in one call then do the string separate.

**Pull request merged**